lib/flat_search_web/plug/local_plug.ex

```diff
@@ -0,0 +1,67 @@
+defmodule FlatSearchWeb.Plugs.Locale do
+  import Plug.Conn
+
+  def init(_opts), do: nil
+
+  def call(conn, _opts) do
+    accepted_languages = extract_accept_language(conn)
+    known_locales = Gettext.known_locales(FlatSearchWeb.Gettext)
+
+    accepted_languages =
+      known_locales --
+        (known_locales -- accepted_languages)
+
+    case accepted_languages do
+      [locale | _] ->
+        Gettext.put_locale(FlatSearchWeb.Gettext, locale)
+
+        conn
+        |> put_session(:locale, locale)
```

Comment:
`put_session(conn, :locale, locale)` - pipe isn't needed here since it's one function